plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -16,57 +16,90 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import memoizeOne from 'memoize-one';
 import { DataRecord } from '@superset-ui/chart';
 import { QueryFormDataMetric } from '@superset-ui/query';
 import { getNumberFormatter, NumberFormats } from '@superset-ui/number-format';
-import { getTimeFormatter } from '@superset-ui/time-format';
+import {
+  getTimeFormatter,
+  smartDateFormatter,
+  getTimeFormatterForGranularity,
+} from '@superset-ui/time-format';
 
-import { TableChartProps, TableChartTransformedProps, DataType } from './types';
+import { TableChartProps, TableChartTransformedProps, DataType, DataColumnMeta } from './types';
 
 const { PERCENT_3_POINT } = NumberFormats;
+const TIME_COLUMN = '__timestamp';
 
 /**
  * Consolidate list of metrics to string, identified by its unique identifier
  */
 function getMetricIdentifier(metric: QueryFormDataMetric) {
   if (typeof metric === 'string') return metric;
-  // even thought `metric.optionName` is more unique, it's not used
+  // even though `metric.optionName` is more unique, it's not used
   // anywhere else in `queryData` and cannot be used to access `data.records`.
   // The records are still keyed by `metric.label`.
-  return metric.label || 'NOT_LABLED';
+  return metric.label || 'NOT_LABELED';
 }
 
-function isTimeColumn(key: string, data: DataRecord[] = []) {
-  return key === '__timestamp' || data.some(x => x[key] instanceof Date);
+function isTimeColumn(key: string) {
+  return key === TIME_COLUMN;
 }
 
-export default function transformProps(chartProps: TableChartProps): TableChartTransformedProps {
-  const {
-    height,
-    width,
-    datasource,
-    formData,
-    queryData,
-    initialValues: filters = {},
-    hooks: { onAddFilter: onChangeFilter = () => {} },
-  } = chartProps;
+const REGEXP_TIMESTAMP_NO_TIMEZONE = /T(\d{2}:){2}\d{2}$/;
+function isTimeType(key: string, data: DataRecord[] = []) {
+  return isTimeColumn(key) || data.some(x => x[key] instanceof Date);
+}
 
-  const {
-    alignPn: alignPositiveNegative = true,
-    colorPn: colorPositiveNegative = true,
-    showCellBars = true,
-    includeSearch = false,
-    pageLength: pageSize = 0,
-    metrics: metrics_ = [],
-    percentMetrics: percentMetrics_ = [],
-    tableTimestampFormat,
-    tableFilter,
-    timeGrainSqla: granularity,
-    orderDesc: sortDesc = false,
-  } = formData;
-  const { columnFormats, verboseMap } = datasource;
-  const { records, columns: columns_ } = queryData.data || { records: [], columns: [] };
+const processDataRecords = memoizeOne(function processDataRecords(data: DataRecord[] | undefined) {
+  if (!data || !data[0] || !(TIME_COLUMN in data[0])) {
+    return data || [];
+  }
+  return data.map(x => {
+    const time = x[TIME_COLUMN];
+    return {
+      ...x,
+      [TIME_COLUMN]:
+        typeof time === 'string' && time.match(REGEXP_TIMESTAMP_NO_TIMEZONE)
```

Comment:
`REGEXP_TIMESTAMP_NO_TIMEZONE.test(time)` is preferred than `time.match(...)` if you are not planning to use the matched elements.

Reply:
I swear I changed this at one point but somehow it got reverted.